src/hooks/usePosts.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { Post } from '../types';
+import * as postsService from '../api/posts';
+
+export const usePosts = (selectedUserId: number = 0) => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const loadSelectedUserPosts = () => {
+    if (!selectedUserId) {
+      return;
+    }
+
+    setErrorMessage('');
+
+    postsService
+      .getUserPosts(selectedUserId)
+      .then(setPosts)
+      .catch(() => setErrorMessage('Something went wrong'))
```

Comment:
Consider providing more detailed error messages or logging the error to help with debugging. The current message 'Something went wrong' is generic and may not provide enough information for troubleshooting.